CMakeLists.txt

```diff
@@ -20,7 +20,36 @@ set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules ${PROJECT_SOURCE_DIR}/
 set(CMAKE_POSITION_INDEPENDENT_CODE 1)
 set(BISON_FLAGS "-Wno-other")
 
-add_compile_options("-Wno-initializer-overrides")
+set(IMPORTANT_WARNINGS
```

Comment:
I would move this block after detection, because for detection those are not relevant and could cause issue.